.github/workflows/test.yml

```diff
@@ -151,237 +154,50 @@ jobs:
-          --scopes cloud-platform \
-          --metadata=google-monitoring-enabled=true,google-logging-enabled=true \
-          --tags zebrad \
-          --zone "${{ env.ZONE }}"
-          sleep 60
-
-      - name: Sync past mandatory checkpoint
-        id: sync-past-checkpoint
-        run: |
-          gcloud compute ssh \
-          sync-past-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
-          --zone ${{ env.ZONE }} \
-          --quiet \
-          --ssh-flag="-o ServerAliveInterval=5" \
-          --command \
-          "docker run -t --name sync-past-checkpoint \
-          --mount type=bind,source=/mnt/disks/gce-containers-mounts/gce-persistent-disks/${{ env.CACHED_DISK_NAME }}-${{ env.GITHUB_SHA_SHORT }},target=/zebrad-cache \
-          ${{ env.GAR_BASE }}/${{ env.IMAGE_NAME }}:sha-${{ env.GITHUB_SHA_SHORT }} \
-          cargo test --locked --release --features enable-sentry,test_sync_past_mandatory_checkpoint_${{ env.NETWORK }} --manifest-path zebrad/Cargo.toml sync_past_mandatory_checkpoint_${{ env.NETWORK }}"
-
-          EXIT_CODE=$(\
-          gcloud compute ssh \
-          sync-past-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} \
-          --zone ${{ env.ZONE }} \
-          --quiet \
-          --ssh-flag="-o ServerAliveInterval=5" \
-          --command="docker wait sync-past-checkpoint")
-
-          exit ${EXIT_CODE}
-
-      - name: Delete test instance
-        # We don't want to leave a failed instance in GCP using resources
-        if: always()
-        continue-on-error: true
-        run: |
-          INSTANCE=$(gcloud compute instances list --filter=sync-past-checkpoint-${{ env.GITHUB_REF_SLUG_URL }}-${{ env.GITHUB_SHA_SHORT }} --format='value(NAME)')
-          if [ -z "${INSTANCE}" ]; then
-            echo "No instance to delete"
-          else
-            gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
-          fi
+    needs: build
+    uses: ./.github/workflows/gcp-test-deploy.yml
+    with:
+      app_name: zebrad
+      test_id: sync-past-checkpoint
+      test_name: Test sync past checkpoint
+      test_description: Test full validation sync from a cached state
+      test_variables: '-e TEST_CHECKPOINT_SYNC=1 -e ZEBRA_FORCE_USE_COLOR=1'
+      zebra_state_path: '/zebrad-cache'
+      disk_prefix: zebrad-cache
+      disk_suffix: checkpoint
+      builds_disks: false
+      needs_constants: false
+
+  # Test that Zebra can run a full mainnet sync after a PR is approved
+  test-full-sync:
+    needs: build
+    uses: ./.github/workflows/gcp-test-deploy.yml
+    with:
+      app_name: zebrad
+      test_id: full-sync-to-tip
+      test_name: Test full sync
+      test_description: Test a full sync up to the tip
+      test_variables: '-e TEST_FULL_SYNC=1 -e ZEBRA_FORCE_USE_COLOR=1 -e FULL_SYNC_MAINNET_TIMEOUT_MINUTES=600'
+      zebra_state_path: '/zebrad-cache'
```

Comment:
This seems common for all right now, is there a way to have a default value in the `workflow_call` inputs that will allow this to be `/zebrad-cache` by default but optionally override-able? I think you can do similar with `workflow_dispatch`

Reply:
Similar for `disk_prefix`

Reply:
Great idea! Yes, this can be done.

https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_callinputs

Reply:
Applied